config/tailwindcss-preset.config.js

```diff
@@ -0,0 +1,196 @@
+  ...StyleDictionary.transformGroup.js,
+  'attribute/cti',
+  'name/cti/kebab2',
+];
+
+/**
+ * Custom formatter for Tailwindcss config preset
+ */
+StyleDictionary.registerFormat({
+  name: 'tailwindcss/preset',
+  formatter: function ({ dictionary, file }) {
+    const tokens = {};
+    const header = StyleDictionary.formatHelpers.fileHeader({ file });
+    deep.p = true;
+
+    dictionary.allTokens
+      .filter((token) => token.path[0] !== 'core')
+      .map(remapConfigKeys)
+      .map(patchSpacingKeys)
+      .forEach((token) => {
+        deep(tokens, token.configKeys, `var(--${token.name})`);
+      });
+
+    const rawSource = `
+      ${header}
+      
+      const plugin = require('tailwindcss/plugin');
+
+      /**
+       * Flatten shadow keys similarly to 
+       * \`flattenColorPalette\` utility function used by color core plugins
+       */
+      const flattenShadows = (shadows) =>
+        Object.assign(
+          {},
+          ...Object.entries(shadows ?? {}).flatMap(([shadow, values]) =>
+            typeof values == 'object'
+              ? Object.entries(flattenShadows(values)).map(([name, value]) => ({
+                  [shadow + '-' + name]: value,
+                }))
+              : [{ [shadow]: values }]
+          )
+        )
+
+      module.exports = 
+        {
+          theme: ${JSON.stringify(tokens)},
+          plugins: [
+            /**
+             * Custom plugin to convert \`text-style\` design token to the \`font\` css shorthand property
+             * as there is no mapping provided by the core plugins.
+             */
+            plugin(function ({ matchUtilities, theme }) {
+              matchUtilities(
+                {
+                  'text-style': (value) => ({
+                    font: value,
+                  }),
+                },
+                { values: theme('textStyle') }
+              );
+            }),
+            /**
+             * Custom plugin to convert \`shadow\` design token to the \`boxShadow\` css shorthand property
+             * and to avoid functionality of core tailwindcss plugins (\`boxShadow\` and \`boxShadowColor\`).
+             */
+             plugin(function ({ matchUtilities, theme }) {
+              matchUtilities(
+                {
+                  shadow: (value) => ({
+                    boxShadow: value,
+                  }),
+                },
+                { values: flattenShadows(theme('shadow')) }
+              );
+            }),
+          ],
+          corePlugins: {
+            boxShadow: false,
+            boxShadowColor: false,
+          },
+        };
+    `;
+
+    return prettier.format(rawSource, {
+      parser: 'babel',
+      singleQuote: true,
+    });
+  },
+});
+
+/**
+ * Configuration of mappings of the start of the original paths
+ * to the generated Tailwindcss config key names
+ */
+const mappings = [
+  { original: ['color', 'text-icon'], tailwindcss: ['colors', 'icon'] },
```

Comment:
I would rename `icon` to `text-&-icon` for correctness, because it could be misleading — or just make it `text`, but I still think the full name is better to keep the semantics (that these colors should be use for text and icons only).

Reply:
Now you end up with:

```js
module.exports = {
  theme: {
    colors: {
      icon: {
        standard: 'var(--telekom-color-text-and-icon-standard)',
        additional: 'var(--telekom-color-text-and-icon-additional)',
        disabled: 'var(--telekom-color-text-and-icon-disabled)',
        …
```

Reply:
Hi,

Yeah, keeping the full name is probably better, even if it is a bit longer:  `icon` -> `text-&-icon`

Converting `standard` keys to `DEFAULT` is also a great suggestion. I have modified the `flattenShadows` function to accept it.